scripts/install/configure-apt-cache.sh

```diff
@@ -1,52 +1,45 @@
 #!/bin/bash -e
 
-# Tell apt where to find the cache for packages, if one has been setup.
-# We look for the APT_PROXY_PORT environment variable to be set to the port
-# number, else for the first argument to be set to the port number.
-
-# Requires root access in order to write to /etc/apt/apt.conf.d/01proxy
-
-# TODO(james): Determine if we can replace this change of the config file
-# with a --option flag on apt-get, ala:
+# Tell apt where to find the local (on-host) caching proxy for APT packages,
+# if one has been setup. Pass in the port as the first (and only) argument
+# to this script, else set the APT_PROXY_PORT environment variable to the
+# port number. Records the URL of the proxy "permanently" in
+# /etc/apt/apt.conf.d/01proxy; root root access is required to write to
+# that file.
+#
+# Using this script is appropriate if it isn't practical to pass the
+# URL of the proxy on the apt-get command line, such as this way:
 #
 #   apt-get -o "Acquire::HTTP::Proxy=${APT_PROXY_URL}" install ...
 #
-# This would avoid the need for sudo outside of docker build, and would
-# avoid docker build permanently recording the proxy in docker image.
-
+# The script install-apt-packages.sh uses this latter approach.
+#
 # To learn more, see:
 # https://github.com/sameersbn/docker-apt-cacher-ng
 # https://gist.github.com/dergachev/8441335
 # https://chandrusoft.wordpress.com/2012/09/30/disable-proxy-when-using-apt-get/
 
-APT_PROXY_PORT="${APT_PROXY_PORT:-${1}}"
-if [ -z "${APT_PROXY_PORT}" ] ; then
-  echo "No apt cache proxy declared via APT_PROXY_PORT or script argument."
-  exit 0
-fi
-echo "APT_PROXY_PORT: ${APT_PROXY_PORT}"
+THIS_DIR="$(dirname "$(readlink -f "${0}")")"
```

Comment:
These nested double quotes don't seem right but I'm assuming this works since it is in all the files here.

Reply:
That's the magic of using $() rather than backticks!